api/src/main/java/com/datatorrent/api/Attribute.java

```diff
@@ -146,6 +146,8 @@ public String toString()
 
     Set<Map.Entry<Attribute<?>, Object>> entrySet();
 
+    <T> T getValue(Attribute<T> key);
```

Comment:
It looks more like a helper or utility method that does not need to bloat the interface. It may be important to know whether a values was set or not, I suppose that's why get(...) does what it does. Also, methods in the interfaces should be documented.